Framework/Core/include/Framework/RawDeviceSource.h

```diff
@@ -0,0 +1,56 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef FRAMEWORK_RAWDEVICESOURCE_H
+#define FRAMEWORK_RAWDEVICESOURCE_H
+
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/OutputSpec.h"
+#include "Framework/DataAllocator.h"
+#include <fairmq/FairMQParts.h>
+#include <vector>
+#include <functional>
+
+namespace o2 {
+namespace framework {
+  using InjectorFunction = std::function<void(FairMQDevice &device, FairMQParts& inputs, int index)>;
+
+  /// Helper function which takes a set of inputs coming from a device,
+  /// massages them so that they are valid DPL messages using @param spec as header
+  /// and sends them to the downstream components.
+  InjectorFunction incrementalConverter(OutputSpec const &spec, uint64_t startTime, uint64_t step);
+
+  /// This is to be used for sources which already have an O2 Data Model /
+  /// (header, payload) structure for their output. At the moment what this /
+  /// does is to add a DataProcessingHeader. In the future, it will hopefully not
+  /// be required.
+  InjectorFunction o2DMAdaptor(OutputSpec const &spec, uint64_t startTime, uint64_t step);
```

Comment:
"abbreviations should be avoided." according to the coding guidelines. I would propose to use "DataModel" instead of "DM".

Reply:
Ok, will fix.